### PR TITLE
when computing the outline/structure of a document consider all frameworks

### PR DESCRIPTION
**Pull request by jrieken**
...ks

### CODE REVIEW

**Comment by jrieken** [comment]
Code as shown below should give a merged view when being asked for the its sturcture. The individual element have to know to what framework they belong

```
{
    public class A
    {
    
        public void M() 
        {
            
        }
#if ASPNETCORE50
        public void N() 
        {
            
        }
#else
        
        public void O() 
        {
            
        }
#endif
    }
}
```

**Comment by nosami** [comment]
Been thinking about this. Would you ever need to get a list for more than one framework at a time?

Would it be better to request the structure for the framework that you are interested in?

**Comment by nosami** [comment]
OK... I get it now. Ignore my last comment!

**Comment by jrieken** [comment]
:-) with my last change the Node has a list of Frameworks. That would allow further client side filtering or special rendering depending on the selected framework.

File: src/OmniSharp/Models/Node.cs [hunk 1]
```diff
@@ -1,13 +1,78 @@
+using System;
 using System.Collections.Generic;
 
 namespace OmniSharp.Models
 {
-    public class Node
+    public class Node : IComparable<Node>
```
Comment (davidfowl) [inline src/OmniSharp/Models/Node.cs]: This class really needs to be renamed. I have no idea what a node is.

**Comment by jrieken** [comment]
:ok_hand: going with a list of project names. Node is the worst name of all times ;-) Will go with ```FileMemberElement```

**Pull request merged**